tests/ha/single_pod_restart/test_data_server_pod_restart.py

```diff
@@ -723,3 +723,422 @@ def test_cont_io_data_server_pod_restart(self):
+                                                    log_prefix=self.test_prefix, skipwrite=True,
+                                                    skipcleanup=True, setup_s3bench=False,
+                                                    nsamples=5, nclients=5)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 13.1: Performed READs/Verify on data written in healthy mode in step-1")
+
+        if CMN_CFG["dtm0_disabled"]:
+            degraded_mode_prefix_list = [self.test_prefix_deg, self.test_prefix_datapod_restart,
+                                         self.test_prefix_server_deg]
+            LOGGER.info("STEP 13.2: Perform READs/Verify on data written after data pod down "
+                        "in step-4, data pod restart in step-8 and server pod down step-11")
+            for test_prefix in degraded_mode_prefix_list:
+                resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                            log_prefix=test_prefix, skipwrite=True,
+                                                            skipcleanup=True, setup_s3bench=False,
+                                                            nsamples=5, nclients=5)
+                assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 13.2: Performed READs/Verify on data written after data pod down "
+                        "in step-4, data pod restart in step-8 and server pod down step-11")
+
+            LOGGER.info("Step 14: Create new IAM user and buckets, Perform WRITEs-READs-Verify "
+                        "with variable object sizes after data pod restart")
+            users = self.mgnt_ops.create_account_users(nusers=1)
+            self.test_prefix = f'test-45510-restart-{int(perf_counter_ns())}'
+            self.s3_clean.update(users)
+        else:
+            LOGGER.info("Step 14: Create new objects, Perform WRITEs-READs-Verify with variable "
+                        "object sizes after data and server pod restart")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix,
+                                                    skipcleanup=True, setup_s3bench=False,
+                                                    nsamples=5, nclients=5)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 14: Performed WRITEs-READs-Verify with variable sizes objects after "
+                    "data and server pod restart")
+        LOGGER.info("COMPLETED: Verify IO when 1 data pod restart and then verify IO when 1 "
+                    "server pod restart by replica method")
+
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-45536")
+    def test_io_datapod_serverpod_rst(self):
+        """
+        Verify IO when 1 data pod and 1 server pod down and then restart by replica
```

Comment:
we can change the docstring to say that restart data pod and run read/verify during server pod restart or similar here as well in started and completed logger